### PR TITLE
Small code improvements

### PR DESCRIPTION
**Pull request by mssalvatore**
# What does this PR do? 

This PR contains a number of small code quality improvements. 

Add any further explanations here. 

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] ~Is the TravisCI build passing?~
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by running locally with the `MockPuppet`
* [ ] ~If applicable, add screenshots or log transcripts of the feature working~

### CODE REVIEW

File: monkey/infection_monkey/master/automated_master.py [hunk 1]
```diff
@@ -195,11 +195,8 @@ def _run_plugins(self, plugin: List[Any], plugin_type: str, callback: Callable[[
         logger.info(f"Running {plugin_type}s")
         logger.debug(f"Found {len(plugin)} {plugin_type}(s) to run")
 
-        for p in plugin:
-            if self._stop.is_set():
-                logger.debug(f"Received a stop signal, skipping remaining {plugin_type}s")
-                return
-
+        interrupted_message = f"Received a stop signal, skipping remaining {plugin_type}s"
+        for p in interruptable_iter(plugin, self._stop, interrupted_message):
```
Comment (VakarisZ) [inline monkey/infection_monkey/master/automated_master.py:199]: Unrelated, but there's a typo: `plugins` not `plugin`

File: monkey/infection_monkey/utils/threading.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    for t in worker_threads:
+        t.join()
+
+
+def create_daemon_thread(target: Callable[..., None], args: Tuple = ()):
+    return Thread(target=target, args=args, daemon=True)
+
+
+def interruptable_iter(
+    iterator: Iterable, interrupt: Event, log_message: str = None, log_level: int = logging.DEBUG
```
Comment (VakarisZ) [inline monkey/infection_monkey/utils/threading.py:24]: There's still a bit of duplication in that all users of this method initialize essentially the same stop event and all pass the same stop event here. Maybe we could have a `StoppableThread` class or something. Also, logging sounds like a different responsibility. But good enough for now

File: monkey/infection_monkey/master/automated_master.py (hunk 1 above)
Comment (mssalvatore) [reply monkey/infection_monkey/master/automated_master.py]: Good catch. Fixed.

File: monkey/infection_monkey/utils/threading.py (hunk 2 above)
Comment (mssalvatore) [reply monkey/infection_monkey/utils/threading.py:24]: We can improve on it as we go. I know what you mean about logging, but I'm not sure of a better way to do it at this time.

The fact that everything is using the same stop event is incidental. For example, it cannot be assumed that the stop event in the `Propagator` and `Ransomware` payload are the same. For the moment they are, but we may want to be able to stop these activities independently in the future.

**Review by VakarisZ — APPROVED** [review]
A typo and a consideration, but certainly good enough for now

**Pull request merged**